sdk/storage/storage-blob/src/credentials/RawTokenCredential.ts

```diff
@@ -26,39 +23,36 @@ import { TokenCredentialPolicy } from "../policies/TokenCredentialPolicy";
  *    }
  *  }, 60 * 60 * 1000); // Set an interval time before your token expired
  * @export
- * @class TokenCredential
- * @extends {Credential}
+ * @implements {TokenCredential}
  *
  */
-export class TokenCredential extends Credential {
+export class RawTokenCredential implements TokenCredential {
   /**
    * Mutable token value. You can set a renewed token value to this property,
    * for example, when an OAuth token is expired.
    *
    * @type {string}
-   * @memberof TokenCredential
    */
   public token: string;
 
   /**
    * Creates an instance of TokenCredential.
    * @param {string} token
-   * @memberof TokenCredential
    */
   constructor(token: string) {
-    super();
     this.token = token;
   }
 
   /**
-   * Creates a TokenCredentialPolicy object.
-   *
-   * @param {RequestPolicy} nextPolicy
-   * @param {RequestPolicyOptions} options
-   * @returns {TokenCredentialPolicy}
-   * @memberof TokenCredential
+   * Retrieves the token stored in this RawTokenCredential.
+   * 
+   * @param _scopes Ignored since token is already known.
+   * @param _options Ignored since token is already known.
    */
-  public create(nextPolicy: RequestPolicy, options: RequestPolicyOptions): TokenCredentialPolicy {
-    return new TokenCredentialPolicy(nextPolicy, options, this);
+  getToken(_scopes: string | string[], _options?: GetTokenOptions): Promise<AccessToken | null> {
+    return Promise.resolve({
+      token: this.token,
+      expiresOnTimestamp: Date.now() + 2 * 60 * 1000 // 2 Minutes
```

Comment:
>expiresOnTimestamp [](start = 6, length = 18)

We should assume token provided in `RawTokenCredential` will never expired, because it's up to customers refreshing token string value.

Reply:
We've implemented token refreshing based on expiration time in `BearerTokenAuthenticationPolicy` so since it's possible for the user to change the `token`, we need a relatively short expiration so we aren't using an out of date token.

One possibility is that I could change that caching code to always ask for the current token when `expiresOnTimestamp` is set to `0`.

Reply:
If am correct, `RawTokenCredential` is a simple wrapper for bearer token string. It needs manually refresh by customers upodating `token` property.

I'm not sure what happens after 2 mins timeout. Because `RawTokenCredential` doesn't define any token refresh logic. 

So what will hapen after 2mins timeout? Will underlayer just call `getToken` again? If so, this call can be optimized by setting up an infinite token..

---
In reply to: [295116310](https://github.com/Azure/azure-sdk-for-js/pull/3853#discussion_r295116310) [](ancestors = 295116310)